### PR TITLE
Provider > Add link API between on-prem VCS and Agent-Pool 

### PR DESCRIPTION
**Pull request by lyzohub**

### CODE REVIEW

File: run_triggers_test.go [hunk 1]
```diff
@@ -88,14 +88,16 @@ func TestRunTriggersRead(t *testing.T) {
...
+	createdTrigger, err := client.RunTriggers.Create(ctx, options)
 	require.NoError(t, err)
-	assert.NotEmpty(t, created_trigger.ID)
+	assert.NotEmpty(t, createdTrigger.ID)
 
 	t.Run("get run trigger by id", func(t *testing.T) {
-		trigger, err := client.RunTriggers.Read(ctx, created_trigger.ID)
+		trigger, err := client.RunTriggers.Read(ctx, createdTrigger.ID)
 		require.NoError(t, err)
-		assert.Equal(t, created_trigger, trigger)
```
Comment (lyzohub) [inline run_triggers_test.go:98]: Comparing memory adresses

File: vcs_provider_test.go [hunk 2]
```diff
@@ -95,6 +95,25 @@ func TestVcsProvidersCreate(t *testing.T) {
 		}
 	})
 
+	t.Run("with agent-pool vcs-enabled: false", func(t *testing.T) {
```
Comment (lyzohub) [inline vcs_provider_test.go:98]: To create test case for vcs-enabled: true scenarion, CI changes needed because of agents online validation

**Review by petroprotsakh — APPROVED** [review]
lgtm

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Some non-required builds are failing or stuck.

**Comment by emocharnik** [comment]
@lyzohub, Required builds are failing.

**Comment by emocharnik** [comment]
@lyzohub, Some non-required builds are failing or stuck.

**Pull request merged**